lib/pub_sub/publisher.rb

```diff
@@ -1,48 +1,55 @@
 module PubSub
   class Publisher
-
-    def self.publish(message, async: false)
-      if async
-        new.publish_asynchronously(message)
-      else
-        new.publish_synchronously(message)
+    class << self
+      def publish(message, async: false)
+        if async
+          publish_asynchronously(message)
+        else
+          publish_synchronously(message)
+        end
       end
-    end
-
-    def self.topic
-      new.topic
-    end
 
-    def publish_asynchronously(message)
-      Thread.new do
-        message_id = publish_synchronously(message)
-        PubSub.logger.info(
-          "Message published asynchronously: #{message_id}"
-        )
+      def publish_asynchronously(message)
+        Thread.new do
+          message_id = publish_synchronously(message)
+          PubSub.logger.info(
+            "Message published asynchronously: #{message_id}"
+          )
+        end
       end
-    end
 
-    def publish_synchronously(message)
-      sns.publish(
-        topic_arn: topic_arn,
-        message: message
-      ).message_id
-    end
+      def publish_synchronously(message)
+        Breaker.current_breaker.run do
+          sns.publish(
+            topic_arn: topic_arn,
+            message: message
+          ).message_id
+        end
+      rescue CB2::BreakerOpen
+        Breaker.use_next_breaker
+        sleep 1
+        retry
+      end
 
-    private
+      private
 
-    def sns
-      @sns ||= Aws::SNS::Client.new
-    end
+      def sns
+        Aws::SNS::Client.new(region: Breaker.current_region)
```

Comment:
I don't believe these are connections - as far as I'm able to determine from reading their library, these clients act to create connections. Rather this is more a matter of setting up the right headers, config, etc.

These don't involve a network call to create, so I don't think we're saving much time by caching them.